src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -672,19 +673,32 @@ def get_active_daemon(self, daemon_descrs: List[DaemonDescription]) -> DaemonDes
         return DaemonDescription()
 
     def fail_over(self) -> None:
-        if not self.mgr_map_has_standby():
-            raise OrchestratorError('Need standby mgr daemon', event_kind_subject=(
-                'daemon', 'mgr' + self.mgr.get_mgr_id()))
-
-        self.mgr.events.for_daemon('mgr' + self.mgr.get_mgr_id(),
-                                   'INFO', 'Failing over to other MGR')
-        logger.info('Failing over to other MGR')
-
-        # fail over
-        ret, out, err = self.mgr.check_mon_command({
-            'prefix': 'mgr fail',
-            'who': self.mgr.get_mgr_id(),
-        })
+        # this has been seen to sometimes transiently fail even when there are multiple
+        # mgr daemons. As long as there are multiple known mgr daemons, we should retry.
+        no_standby_exc = OrchestratorError('Need standby mgr daemon', event_kind_subject=(
+            'daemon', 'mgr' + self.mgr.get_mgr_id()))
+        if len([d for d in self.mgr.cache.get_daemons_by_type('mgr')]) < 2:
+            # only one known mgr daemon, retry won't help us
+            raise no_standby_exc
```

Comment:
seems like this could also cause transient upgrade failures, maybe the check during `mgr_map_has_standby` is sufficient?

Reply:
my thinking was, since this was appearing during upgrade and we already have a hard check for 2 mgr daemons marked running in the cache there then it must not be what we have cached that's the issue. However, thinking about it, those checks existing also makes this redundant anyway so I should really just remove it anyhow.